### ISSUE TITLE
Support for sitemap rel

### ISSUE THREAD

**Issue by nranas** [cignium/hypermedia-client#24]
## Feature

We need to support the new rel: `sitemap`. A typical response from performing a `GET` on that URL could look like this:

``` json
[
  {
    "type": "string",
    "value": "Step 1",
    "links": [
      {
        "href": "http://...",
        "rel": "navigate"
      }
    ]
  },
  {
    "value": "Step 2",
    "type": "string",
    "errors": [
      "User must have a name!"
    ],
    "warnings": [
      "Form have unsaved changes"
    ],
    "links": [
      {
        "href": "http://...",
        "rel": "navigate"
      }
    ]
  },
  {
    "title": "Step 3",
    "type": "array",
    "links": [
      {
        "href": "http://...",
        "rel": "navigate"
      }
    ],
    "items": [
      {
        "type": "string",
        "value": "Step 3.1",
        "links": [
          {
            "href": "http://...",
            "rel": "navigate"
          }
        ]
      },
      {
        "type": "string",
        "value": "Step 3.2",
        "links": [
          {
            "href": "http://...",
            "rel": "navigate"
          }
        ]
      },
      {
        "type": "string",
        "value": "Step 3.3",
        "links": [
          {
            "href": "http://...",
            "rel": "navigate"
          }
        ]
      }
    ]
  }
]
```

The response in this example should be rendered like so at the top of the page with links for each navigate-link:

| Step 1 | Step 2 | Step 3 |
| --- | --- | --- |
|  |  | Step 3.1 |
|  |  | Step 3.2 |
|  |  | Step 3.3 |

**Comment by hfjallemark** [comment]
It can return any response, right? As long as it is following the media type. We won't render it any differently from a normal document so theoretically you could have form elements in it as well.

**Comment by nranas** [comment]
Don't we want to render it differently? I thought we wanted a horizontal layout. Or are you saying that is strictly in the default styling of the CSS-classes added on this component?

**Comment by hfjallemark** [comment]
I don't think we want to render it any differently, but by wrapping it within a `ct-sitemap` class it could be styled in whichever way it needs to be.

**Comment by nranas** [comment]
Right, I updated the comment.

**Comment by nranas** [comment]
Solved by #25
